### PR TITLE
Update constructor example in generic-odata-client.mdx

### PR DESCRIPTION
**Pull request by andrastemesi-sap**
## What Has Changed?

Update constructor example in generic-odata-client.mdx 

## Manual Checks?

- [x] Text adheres to the style guide (`vale docs/`)
  - Every sentence is on its own line
  - Headings use [title capitalization](https://capitalizemytitle.com/style/AP/#) (applies also to the sidebar and title of a document)
  - You followed [naming center](https://www.sapbrandtools.com/naming-center/#/dashboard) guidelines when referring to SAP products (e.g. SAP S/4HANA)
- [x] You checked your spelling and grammar (consider using Grammarly, see `CONTRIBUTING.md`)
- [x] You formatted all changed files with prettier (`npm run prettier`)
- [x] You tested if the documentation still builds (`npm run build`)
- [x] You verified all new and changed links still work (changing the `id` or name of a file can break links)
- [x] You have updated the [feature matrix](https://github.com/SAP/cloud-sdk/blob/main/docs/components/data/features.js) if you add documentation on a new feature or otherwise required.

### CODE REVIEW

File: docs/java/features/odata/generic-odata-client.mdx [hunk 1]
```diff
@@ -46,13 +46,13 @@ Requests are represented by dedicated request classes for each operation.
 For example to read a collection of entities use:
 
 ```java
-new ODataRequestRead(servicePath, entityName, encodedQuery, ODataProtocol.V4);
+new ODataRequestRead(servicePath, structuredQuery);
```
Comment (newtork) [inline docs/java/features/odata/generic-odata-client.mdx:49]: As a reader, what does `structuredQuery` mean?
- which type?
- how to instantiate?

Comment (andrastemesi-sap) [reply docs/java/features/odata/generic-odata-client.mdx:49]: next line is a tip that I modified:

> Structured queries can be created conveniently as described in [Building Queries below](#building-queries).

File: docs/java/features/odata/generic-odata-client.mdx [hunk 2]
```diff
@@ -46,13 +46,14 @@ Requests are represented by dedicated request classes for each operation.
...
```
Comment (MatKuhr) [inline docs/java/features/odata/generic-odata-client.mdx:49]:
```suggestion
```

File: docs/java/features/odata/generic-odata-client.mdx [hunk 3]
```diff
@@ -46,13 +46,14 @@ Requests are represented by dedicated request classes for each operation.
...
 // e.g.
```
Comment (MatKuhr) [inline docs/java/features/odata/generic-odata-client.mdx:50]:
```suggestion
```

File: docs/java/features/odata/generic-odata-client.mdx [hunk 4]
```diff
@@ -46,13 +46,14 @@ Requests are represented by dedicated request classes for each operation.
...
-new ODataRequestRead("/path/to/service", "A_Entity", "$select=Foo%20Bar", ODataProtocol.V4);
+StructuredQuery query = StructuredQuery.onEntity("Movies", ODataProtocol.V4);
```
Comment (MatKuhr) [inline docs/java/features/odata/generic-odata-client.mdx:51]:
```suggestion
StructuredQuery query = StructuredQuery.onEntity("A_Entity", ODataProtocol.V4);
```

**Review by MatKuhr — APPROVED** [review]

**Comment by MatKuhr** [comment]
Let's wait with a merge until the feature is released

**Comment by artemkovalyov** [comment]
@MatKuhr, can this be merged now?

**Review by MatKuhr — APPROVED** [review]

**Pull request merged**